app/src/main/java/vn/sunasterisk/music_67/ui/detailplaying/DetailPlayingActivity.kt

```diff
@@ -1,39 +1,207 @@
 package vn.sunasterisk.music_67.ui.detailplaying
 
-import android.os.Bundle
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.os.*
 import android.support.v7.app.AppCompatActivity
+import android.view.View
+import android.widget.SeekBar
 import com.bumptech.glide.Glide
 import com.bumptech.glide.request.RequestOptions
+import kotlinx.android.synthetic.main.activity_detail_playing.*
 import vn.sunasterisk.music_67.R
-import vn.sunasterisk.music_67.ui.genres.DEFAULT_VALUE
+import vn.sunasterisk.music_67.data.model.Track
+import vn.sunasterisk.music_67.service.ACTION_CREATE
+import vn.sunasterisk.music_67.service.PlayingSet
+import vn.sunasterisk.music_67.service.PlayingTracksService
+import vn.sunasterisk.music_67.service.TrackStateListener
 import vn.sunasterisk.music_67.utils.TimeConvert
 import vn.sunasterisk.music_67.utils.TrackAttributes
-import kotlinx.android.synthetic.main.activity_detail_playing.*
 
-class DetailPlayingActivity : AppCompatActivity() {
+private const val MSG_HANDLER = 1
+
+class DetailPlayingActivity : AppCompatActivity(), View.OnClickListener, TrackStateListener {
+	private lateinit var playingTracksService: PlayingTracksService
+	lateinit var handler: Handler
+	private lateinit var thread: Thread
+	private var isBound = true
+	private val serviceConnection = object : ServiceConnection {
+		override fun onServiceDisconnected(name: ComponentName?) {
+			isBound = false
+		}
+
+		override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+			val binder = service as PlayingTracksService.LocalBinder
+			playingTracksService = binder.getServive()
+			getState()
+			updateState()
+			isBound = true
+			playingTracksService.addListener(this@DetailPlayingActivity)
+		}
+	}
+
 	override fun onCreate(savedInstanceState: Bundle?) {
 		super.onCreate(savedInstanceState)
 		setContentView(R.layout.activity_detail_playing)
 		getContentIntent()
+		playingTracksService = PlayingTracksService()
+	}
+
+	override fun onStart() {
+		super.onStart()
+		val playSongIntent = Intent(this, PlayingTracksService::class.java)
+		playSongIntent.apply {
+			action = ACTION_CREATE
+		}
+		startService(playSongIntent)
+		bindService(playSongIntent, serviceConnection, Context.BIND_AUTO_CREATE)
+		PlayingSet.isStartActivity = true
+		pauseToPlay()
+	}
+
+	override fun onResume() {
+		super.onResume()
+		imagePlay.setOnClickListener {
```

Comment:
Tại sao lại setOnClick Ở đây.
Em nên set trong onViewCreated
Ex:
` override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        registerListeners()
    }`
`fun registerListeners() {
        imagePlay.setOnClickListener(this)
        imageNext.setOnClickListener(this)
    }`